File: Programmers/pro_17682_다트 게임.py
def solution(dartResult):
    stack = []
    cur =""
    for i in dartResult:
        if(i=="*"):
            if(len(stack)>=2):
                stack[-2]*=2
            if(len(stack)>=1):
                stack[-1]*=2
        elif(i=="#"):
            if(stack and stack[-1]>0):
                stack[-1]*=-1
        elif(i.isdigit()):
            cur+=i
        else:
            stack.append(int(cur))
            cur=""
            if(i=="S"):
                continue
            if(i=="D"):
                stack[-1]**=2
            if(i=="T"):
                stack[-1]**=3      
       
    return sum(stack)

File: Programmers/pro_17682_다트게임.js
function solution(dartResult) {
    const score = { "S": 1, "D": 2, "T": 3 }
    let cur = ''
    const stack = []

    for (let i = 0, len = dartResult.length; i < len; i++) {
        let char = dartResult[i]
        let lenS = stack.length
        if ('A' <= char) {
            stack.push(Math.pow(cur, score[char]))
            cur = ''
        } else if (char === '*') {
            if (lenS >= 2)
                stack[lenS - 2] *= 2
            if (lenS)
                stack[lenS - 1] *= 2
        } else if (char === '#' && lenS) {
            stack[lenS - 1] *= -1
        } else {
            cur += char
        }
    }
    return stack.reduce((acc, v) => acc + v);
}